test/integration/apiserver.go

```diff
@@ -471,6 +472,31 @@ func offeringService(ctx context.Context, conn *grpc.ClientConn, managementClien
 			assert.Equal(t, expectedResult, offering)
 			return true, nil
 		}, offeringCtx.Done()))
+
+		// watch offerings
+		watchClient, err := client.Watch(offeringCtx, &apiserverv1.OfferingWatchRequest{
+			Account: testName,
+		})
+		require.NoError(t, err)
+		counter := 0
+		expectedEventNum := 2
+		assert.NoError(t, wait.PollUntil(time.Second, func() (done bool, err error) {
+			event, err := watchClient.Recv()
```

Comment:
This is a blocking call, thus no need for such polling complications. 

https://grpc.io/docs/tutorials/basic/go/#server-side-streaming-rpc-1

Reply:
Also, read the docs, both `RecvMsg` and `SendMsg` are blocking calls. The timeout is injected via the original context.

Reply:
done